### PR TITLE
Github Actions: Debian package updater

### PR DESCRIPTION
**Pull request by theofficialgman**
this is a complex helper script designed to make automatically updating packages from online debian repos easy. simply set a couple of variables in your install/install-32/install-64 script and create a matching script in the updates folder and let the github actions take care of the rest

I have added the new "Syncthing" app to showcase this script (not added to the categories list yet). I have tested the github actions and you can see the run of it here https://github.com/theofficialgman/pi-apps/pull/5.
As with the `update_github_script.sh` helper script, you can specify multiple `webPackages` and `packagename` in arrary form and the script will check each for updates.

Documentation for both `update_github_script.sh` and `update_debian_repo_script.sh` will come at some point in the future.

closes: https://github.com/Botspot/pi-apps/issues/1620

### CODE REVIEW

**Comment by theofficialgman** [comment]
note, the filepaths variables in Syncthing are specifically wrong and set to 
```
filepath_32="junk"
filepath_64="junk"
```
to show off the github actions script working on the first run

**Comment by theofficialgman** [comment]
Also, you may wonder why use this for debian repos over the current `update_github_script.sh`.
This script is immune to changes in `Filepath:` on the debian repo (this can happen when repos are restructuring or updating versions, I've seen it on archive.raspberrypi.org even.)

We also don't have to maintain redundant paths in two locations. With the `update_github_script.sh` script format, it requires that the majority of the URL be specified in both the update script as well as the install script. This is not the case here, simply give the location of the debian `Packages` file on the repo, as well as the desired `package` name from that repo, and the script handles sorting all the versions available, and creating the full URL. It also verifies the URL is valid as a fallback as with the other script.

File: .github/workflows/update_debian_repo_script.sh [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+if [ -n "$armhf_webPackages" ] || [ -n "$arm64_webPackages" ] || [ -n "$all_webPackages" ]; then
+#iterate through all variables supplied if an array
+# get longest arrary of passed variables
+num=$(echo -e "${#armhf_webPackages[@]}\n${#arm64_webPackages[@]}\n${#all_webPackages[@]}" | sort -n -r | head -1)
+
+for (( iter=0; iter<$num; iter++ )); do
+#make array appear as one variable inside for loop
+armhf_webPackages=${armhf_webPackages[$iter]}
+arm64_webPackages=${arm64_webPackages[$iter]}
+all_webPackages=${all_webPackages[$iter]}
```
Comment (Botspot) [inline .github/workflows/update_debian_repo_script.sh:20]: I was not aware this was possible: to reuse the same name for both an array and a variable, without encountering complexities. Where did you learn about this, @theofficialgman?

File: .github/workflows/update_debian_repo_script.sh [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+arm64_packagename=${arm64_packagename[$iter]}
+all_packagename=${all_packagename[$iter]}
+if [[ $iter == "0" ]]; then
+    version_number=""
+else
+    version_number=$((iter + 1))
+fi
+
+#set pi_apps_filepath variables
+if [ -n "$armhf_webPackages" ] && [ -f install-32 ]; then
```
Comment (Botspot) [inline .github/workflows/update_debian_repo_script.sh:31]: > `[ -f install-32 ]; then`

What would happen if the working directory was not set to the app folder?

Comment (theofficialgman) [reply .github/workflows/update_debian_repo_script.sh:31]: there would be issues if it wasn't setup as the app folder but now it is setup as the app folder and has been.

the `update_github_script.sh` script makes this same assumption

File: .github/workflows/update_debian_repo_script.sh (hunk 1 above)
Comment (theofficialgman) [reply .github/workflows/update_debian_repo_script.sh:20]: I'm taking advantage of how bash uses variables and arrays.
the array doesn't get overwritten, only the first value in the array gets overwritten.

when you call the array without specifying the index, its assumes the first always.

so what is happening is I overwrite the first index of the array every time I move through the loop with the desired content of a later index

**Comment by theofficialgman** [comment]
I went ahead and merged this since it won't get used unless its included in an apps updater
I do feel like it is a useful addition and different enough from the github/generic updater script

**Pull request merged**